Guide.docc/RuntimeBehavior.md

```diff
@@ -0,0 +1,207 @@
+`self` of the actor, those tasks are effectively going to run on the global concurrent
+pool, and not on the actor. This behavior may be unexpected, but it is the current semantics.
+
+We can correct this a bit more in order to ensure the enqueue order, by isolating
+the tasks to the actor, this is done as soon as we capture the `self` of the actor:
+
+```swift
+// ✅ enqueue order of tasks is guaranteed
+// 
+// ⚠️ however. due to priority escalation, still any order of prints is possible (!) 
+actor Printer {}
+    func go() { // assume this method must be synchronous
+        // Notice the tasks do capture self
+        Task { self.log(1) }
+        Task { self.log(2) }
+        Task { self.log(3) }
+    }
+    
+    func log(_ int: Int) { print(int) }
+}
+```
+
+This improves the situation because the tasks are now isolated to the printer
+instance actor (by means of using `Task{}` which inherits isolation, and refering
+to the actor's `self`), however their specific execution order is _still_ not deterministic.
+
+Actors in Swift are **not** strictly FIFO ordered, and tasks
+processed by an actor may be reordered by the runtime for example because
+of _priority escalation_.
+
+**Priority escalation** takes place when a low-priority task suddenly becomes
+await-ed on by a high priority task. The Swift runtime is able to move such
+task "in front of the queue" and effectively will process the now priority-escalated
+task, before any other low-priority tasks. This effectively leads to FIFO order
+violations, because such task "jumped ahead" of other tasks which may have been
+enqueue on the actor well ahead of it. This does does help make actors very
+responsive to high priority work which is a valuable property to have!
+
+> Note: Priority escalation is not supported on actors with custom executors.
+
+The safest and correct way to enqueue a number of items to be processed by an actor,
+in a specific order is to use an `AsyncStream` to form a single, well-ordered
+sequence of items, which can be emitted to even from synchronous code.
+And then consume it using a _single_ task running on the actor, like this:
+
+```swift
+// ✅ Guaranteed order in which log() are invoked,
+//    regardless of priority escalations, because the disconnect 
+//    between the producing and consuming task
+actor Printer {
+    let stream: AsyncStream<Int>
+    let streamContinuation: AsyncStream<Int>.Continuation
+    var streamConsumer: Task<Void, Never>!
+
+    init() async {
+        let (stream, continuation) = AsyncStream.makeStream(of: Int.self)
+        self.stream = stream
+        self.streamContinuation = continuation
+
+        // Consuming Option A)
+        // Start consuming immediately, 
+        // or better have a caller of Printer call `startStreamConsumer()`
+        // which would make it run on the caller's task, allowing for better use of structured concurrency.
+        self.streamConsumer = Task { await self.consumeStream() }
```

Comment:
Can we please not put spawning an unstructured task in this example here. Instead let's add a `func run() async`  that consumes the stream. The users of this actor can then decide on what task to call that `run` method on.

Reply:
It's explained below. I can inverse the order I guess